src/reflect/value.go

```diff
@@ -217,7 +217,15 @@ func (v Value) CanAddr() bool {
 }
 
 func (v Value) Addr() Value {
-	panic("unimplemented: (reflect.Value).Addr()")
+	if !v.CanAddr() {
+		panic("taking addr of un-addr-able value")
+	}
+
+	return Value{
+		typecode: pointerTo(v.typecode),
+		value:    unsafe.Pointer(&v.value),
+		flags:    v.flags,
```

Comment:
I think this is the bug you mentioned on Slack? Because I think you need to clear the indirect flag here:

```suggestion
		flags:    v.flags &^ valueFlagIndirect,
```